From 5ed21abbef4d4c2983e70bd2868fb817150e883e Mon Sep 17 00:00:00 2001
From: Armin Kuster <akuster808@gmail.com>
Date: Sat, 3 Oct 2020 11:26:46 -0700
Subject: [PATCH] Revert "profiles: Update 'make check' to select tools based
 on USE_SYSTEM"

This reverts commit 6016f931ebf7b61e1358f19453ef262d9d184a4e.

Upstream-Statue: OE specific
These changes cause during packaging with perms changing.

Signed-off-by: Armin Kuster <akuster808@gmail.com>

---
 profiles/Makefile | 50 ++++++++++-------------------------------------
 1 file changed, 10 insertions(+), 40 deletions(-)

diff --git a/profiles/Makefile b/profiles/Makefile
index ba47fc16..5384cb05 100644
--- a/profiles/Makefile
+++ b/profiles/Makefile
@@ -35,49 +35,9 @@ EXTRAS_SOURCE=./apparmor/profiles/extras/
 SUBDIRS=$(shell find ${PROFILES_SOURCE} -type d -print)
 TOPLEVEL_PROFILES=$(filter-out ${SUBDIRS}, $(wildcard ${PROFILES_SOURCE}/*))
 
-ifdef USE_SYSTEM
-    PYTHONPATH=
-    PARSER?=apparmor_parser
-    LOGPROF?=aa-logprof
-else
-    # PYTHON_DIST_BUILD_PATH based on libapparmor/swig/python/test/Makefile.am
-    PYTHON_DIST_BUILD_PATH = ../libraries/libapparmor/swig/python/build/$$($(PYTHON) -c "import distutils.util; import platform; print(\"lib.%s-%s\" %(distutils.util.get_platform(), platform.python_version()[:3]))")
-    LIBAPPARMOR_PATH=../libraries/libapparmor/src/.libs/
-    LD_LIBRARY_PATH=$(LIBAPPARMOR_PATH):$(PYTHON_DIST_BUILD_PATH)
-    PYTHONPATH=../utils/:$(PYTHON_DIST_BUILD_PATH)
-    PARSER?=../parser/apparmor_parser
-    # use ../utils logprof
-    LOGPROF?=LD_LIBRARY_PATH=$(LD_LIBRARY_PATH) PYTHONPATH=$(PYTHONPATH) $(PYTHON) ../utils/aa-logprof
-endif
-
 # $(PWD) is wrong when using "make -C profiles" - explicitely set it here to get the right value
 PWD=$(shell pwd)
 
-.PHONY: test-dependencies
-test-dependencies: __parser __libapparmor
-
-
-.PHONY: __parser __libapparmor
-__parser:
-ifndef USE_SYSTEM
-	@if [ ! -f $(PARSER) ]; then \
-		echo "error: $(PARSER) is missing. Pick one of these possible solutions:" 1>&2; \
-		echo "  1) Test using the in-tree parser by building it first and then trying again. See the top-level README for help." 1>&2; \
-		echo "  2) Test using the system parser by adding USE_SYSTEM=1 to your make command." 1>&2; \
-		exit 1; \
-	fi
-endif
-
-__libapparmor:
-ifndef USE_SYSTEM
-	@if [ ! -f $(LIBAPPARMOR_PATH)libapparmor.so ]; then \
-		echo "error: $(LIBAPPARMOR_PATH)libapparmor.so is missing. Pick one of these possible solutions:" 1>&2; \
-		echo "  1) Build against the in-tree libapparmor by building it first and then trying again. See the top-level README for help." 1>&2; \
-		echo "  2) Build against the system libapparmor by adding USE_SYSTEM=1 to your make command." 1>&2; \
-		exit 1; \
-	fi
-endif
-
 local:
 	for profile in ${TOPLEVEL_PROFILES}; do \
 		fn=$$(basename $$profile); \
@@ -109,6 +69,16 @@ else
   Q=
 endif
 
+ifndef PARSER
+# use system parser
+PARSER=../parser/apparmor_parser
+endif
+
+ifndef LOGPROF
+# use ../utils logprof
+LOGPROF=PYTHONPATH=../utils $(PYTHON) ../utils/aa-logprof
+endif
+
 .PHONY: docs
 # docs: should we have some here?
 docs:
-- 
2.17.1

